lib/logstash/outputs/scalyr.rb

```diff
@@ -368,11 +427,16 @@ def build_multi_event_request_array(logstash_events)
       # optionally set thread
       if origin
         scalyr_event[:thread] = thread_id.to_s
+        scalyr_event[:log] = logs_ids[log_identifier]
```

Comment:
Could you help me understand what this is for.

Reply:
This is setting the reference in the log line event to any log level attributes in the overall addEvents request. In the actual request we use monotonically increasing integers to give IDs to our log attributes instead of the full "serverHost + logfile" combo, the `log_ids` hash is the mapping from the full combo to the integer.